component/azstorage/block_blob.go

```diff
@@ -781,7 +781,11 @@ func (bb *BlockBlob) ReadInBuffer(name string, offset int64, len int64, data []b
 	blobURL := bb.Container.NewBlobURL(filepath.Join(bb.Config.prefixPath, name))
 	opt := bb.downloadOptions
 	opt.BlockSize = len
-	err := azblob.DownloadBlobToBuffer(context.Background(), blobURL, offset, len, data, opt)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
```

Comment:
Any reason to use this. SDK should add this timeout value, right?
If we do go with using the context with timeout, should we update the other calls where `context.Background()` is used.

Reply:
In case of throttling or network disruption, shouldn't we wait for the retries to complete?

Reply:
With the current SDK, I see the value of timeout as 901 seconds in the request URL.
`GET https://storageAccount.blob.core.windows.net/container/a.txt?timeout=901`

In my opinion using 3 mins might not be sufficient for retries (with exponential backoff).